### PR TITLE
Add operator deploymentconfig and imagestream for continuous deployment

### PR DESCRIPTION
**Pull request by dipak-pawar**
*Changes Proposed in this PR*
- Adds ImageStream which points to image from openshift CI cluster.
- Add DeploymentConfig which uses above ImageStream
- Fix minor make targets

To Do
- [ ] Add Documentation
- [ ] Testing on actual clusters

### CODE REVIEW

**Comment by nurali-techie** [comment]
@dipak-pawar the name `operator_dev_cluster.yaml` suggest this is specifically for `dev_cluster` .. but I guess it's not specific to any cluster and can be applied to any cluster !!

File: deploy/operator_dev_cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
```
Comment (xcoulon) [inline deploy/operator_dev_cluster.yaml]: aren't we supposed to have a `DeploymentConfig` that uses this `ImageStream`?

**Comment by xcoulon** [comment]
> @dipak-pawar the name `operator_dev_cluster.yaml` suggest this is specifically for `dev_cluster` .. but I guess it's not specific to any cluster and can be applied to any cluster !!

for now the target is the `dev` clusters since we want to continuously deploy on this environment, as soon as a new Docker Image was pushed on the container registry. For the staging and production environments, the deployment policy may be slightly different

**Comment by dipak-pawar** [comment]
> the name operator_dev_cluster.yaml suggest this is specifically for dev_cluster .. but I guess it's not specific to any cluster and can be applied to any cluster !!

@nurali-techie  yes, you can apply it to any cluster, but not to any production cluster. At this point, we are just using it with dev cluster, this is why I have given this name. I don't think I'll use this imagestream setup for my local developement where I can directly use container image

File: deploy/operator_dev_cluster.yaml (hunk 1 above)
Comment (dipak-pawar) [reply deploy/operator_dev_cluster.yaml]: we are using it here https://github.com/codeready-toolchain/host-operator/pull/38/files#diff-caad408dc9a5d42c733a0e159f579e64R18

**Comment by nurali-techie** [comment]
@dipak-pawar @xcoulon fine :+1:  thanks for clarification :-)

File: deploy/operator_dev_cluster.yaml (hunk 1 above)
Comment (xcoulon) [reply deploy/operator_dev_cluster.yaml]: ah sorry, I missed that  🤦‍♂

**Comment by dipak-pawar** [comment]
thank you @xcoulon  for the review. merging this as wanted to see actual rollout of deployment  in https://console-openshift-console.apps.host.crt-stage.com once we have new image in openshift ci image registry.

**Pull request merged**